lib/assert.js

```diff
@@ -90,7 +90,13 @@ x.throws = function (fn, err, msg) {
 	}
 
 	try {
-		assert.throws(fn, err, msg);
+		if (typeof err === 'string') {
+			assert.throws(fn, function (e) {
+				return e.message === err;
+			}, msg);
+		} else {
+			assert.throws(fn, err, msg);
+		}
```

Comment:
Definitely cleaner, but then I guess it should be like this.

```js
if (typeof err === 'string') {
  var e = err;
  err = function (err) {
    return err.message === e;
  };
}

assert.throws(fn, err, msg);
```

On your code example, at the time the function is executed, `err` is not the string anymore but it is the function. So it will check if `err.message` is equal to the function instead of to the original string.